modules/optimization/doc/content/modules/optimization/getting_started/InvOptTheory.md

```diff
@@ -0,0 +1,376 @@
+    	+\int_{\Gamma_D}\lambda\left(\kappa\nabla\delta T\cdot\boldsymbol{n}\right)~\text{d}\Gamma.
+\end{aligned}
+\end{equation}
+
+Stationarity of $\mathcal{L}$ would require $\delta\mathcal{L} = 0$ for all admissible $\delta T$. Setting each of the integrals in [eqn:delta_Lagrangian] results in the adjoint problem and its boundary conditions
+\begin{equation} \label{eqn:adjoint_problem}
+\boxed{
+\begin{aligned}
+\nabla\cdot \kappa \nabla \lambda +  \sum_{i=1}^{N} \delta (x - x_i)(T - \bar{T}) &=0, ~\text{in}~\Omega,\\
+\lambda  &= 0, ~\text{on}~\Gamma_D, \\
+\kappa \nabla \lambda \cdot \boldsymbol{n} & = \lambda h, ~\text{on}~{\Gamma_R}. \\
+\end{aligned}
+}
+\end{equation}
+Solving [eqn:adjoint_problem] comes down to adjusting the boundary conditions and load vector from the forward problem and re-solving.
+
+[comment2]: <> (% ----------------------------------------------------------------------------------------------------------%)
+
+## PDE Derivatives for Inversion id=sec:PDEDerivs
+
+In this section we will present derivatives for steady state heat conduction [eqn:optimization_problem] with respect to the force or material parameters.  For all of these examples, measurement data is taken at specific locations where the objective function can be represented by [eqn:objective_integral].  We will present the discrete forms of our PDE and its derivative which most closely matches the implementation that will be used in MOOSE.
+
+The discrete form of the PDE constraint for steady state heat conduction in [eqn:optimization_problem], $\hat{g}$, is given as
+
+\begin{equation}\label{eq:discretePDE}
+\hat{\text g}=\textbf{K}\hat{\text T} - \hat{\text f}=0,
+\end{equation}
+where $\textbf{K}$ is the Jacobian matrix, $\hat{\text T}$ and $\hat{\text f}$ are the discretized  temperature and residual vectors. Element-wise definitions of the terms are
+\begin{equation} \label{eq:elementwise_discretized_terms}
+\begin{aligned}
+    \textbf{K}^{\alpha\beta}&=\sum_{e}\int\nabla^\top \text{N}^{\alpha} \cdot \kappa\cdot\nabla\text{N}^{\beta}~\text{d}\Omega,\\
+    \hat{\text f}^{\alpha} & = \sum_{e} \int N^{\alpha} f_b \text{d}\Omega + \sum_{e} \int N^{\alpha}G(T) ~\text{d}\Gamma_R,
+\end{aligned}
+\end{equation}
+where $N^{\alpha}$ denotes the finite element shape function at node $\alpha$. The solution can be expressed as $T(x)\approx \sum_{\alpha} \text{N}^{\alpha}\hat{\text T}^{\alpha} = N \hat{\text T}$. Note here the $\hat{\text f}$ includes the contribution from both the body load term ($f_b$ in [eqn:optimization_problem]) and boundary conditions (see [eqn:optimization_bcs]). We are assuming a Galerkin form for our discretized PDE by making our test and trial functions identical (both are $\text{N}^{\alpha}$). Note in the rest of this document, we omit the superscript of the shape function (i.e., $(\cdot)^{\alpha}$) and the summation over all the elements (i.e., $\sum_e$) for simplicity.
```

Comment:
`the finite element shape function at node $\alpha$` is correct when the shape function is Lagrange type in continuous finite element method.

Reply:
done